src/api/base.ts

```diff
@@ -0,0 +1,59 @@
+import ky from 'ky';
+
+import { ERROR_MESSAGE } from '../utils/constants';
+import { ApiError } from '../types';
+
+// const apiUrl = process.env.NEXT_PUBLIC_API_URL;
+const apiUrl = 'https://sportujspolu-api.onrender.com/api/v1/';
+
+export const api = ky.create({
+  prefixUrl: apiUrl,
+  headers: {},
+});
+
+export const privateApi = (token: string) =>
+  ky.create({
+    prefixUrl: apiUrl,
+    headers: {
+      Authorization: `Bearer ${token}`,
+    },
+  });
```

Comment:
We need to do this somewhere globally and once. Something along the lines of calling `setupKyContext()` in the _app and then reusing the same instance like so:

<img width="561" alt="CleanShot 2023-10-10 at 17 34 03@2x" src="https://github.com/GLObus303/sportujspolu/assets/12488648/8cb4eee6-55cb-45b0-8de9-bb75d7e62357">

you can see the exported instance.

This is other (this time next.js) project. This is in middle of _app render
<img width="457" alt="CleanShot 2023-10-10 at 17 34 59@2x" src="https://github.com/GLObus303/sportujspolu/assets/12488648/dcd8ce15-d4fc-4df2-bc2e-ba104812a1f6">

then function is used for it

<img width="685" alt="CleanShot 2023-10-10 at 17 35 30@2x" src="https://github.com/GLObus303/sportujspolu/assets/12488648/c7f7c9a2-901c-4b4c-ba39-5b56870cc06c">

the instance is simply a variable
<img width="440" alt="CleanShot 2023-10-10 at 17 35 43@2x" src="https://github.com/GLObus303/sportujspolu/assets/12488648/b0c9ccb4-5d07-4ae5-a1e4-9bb39fa89498">